src/Umbraco.Web.UI.Client/src/common/services/tinymce.service.js

```diff
@@ -332,7 +347,35 @@ function tinyMceService($rootScope, $q, imageHelper, $locale, $http, $timeout, s
                     images_replace_blob_uris: false,
                     init_instance_callback: initEvents
                 };
-
+                
+                // We keep spans here, cause removing spans here also removes b-tags inside of them, instead we strip them out later.
+                var validElements = "-strong/b,-em/i,-u,-span,-p,-ol,-ul,-li,-p/div,-a[href|name],sub,sup,strike,br,del,table[width],tr,td[colspan|rowspan|width],th[colspan|rowspan|width],thead,tfoot,tbody,img[src|alt|width|height],ul,ol,li,hr,pre,dl,dt,figure,figcaption,wbr"
+                  
+                // add valid elements from styleFormats.
+                var style, i = 0;
+                for(; i < styles.styleFormats.length; i++) {
+                    style = styles.styleFormats[i];
+                    if(style.block) {
+                        validElements += "," + style.block;
+                    }
+                }
+                
+                var pasteConfig = {
+                    
+                    paste_remove_styles: true,
```

Comment:
Find out what these paste_ properties do and give an inline comment, so we understand whats going on in 6months time when we come back to this code.